src/CoreWCF.Primitives/src/CoreWCF/Channels/Message.cs

```diff
@@ -1056,6 +1055,7 @@ protected override void OnWriteBodyContents(XmlDictionaryWriter writer)
 
         public override async Task OnWriteMessageAsync(XmlDictionaryWriter writer)
         {
+            WriteMessagePreamble(writer);
```

Comment:
This is the bug, yes? Or was there something wrong with using the async methods on XmlWriter?

Reply:
Yes. While most concrete implementations of XmlWriter support using the Async methods, XmlDictionaryWriter implementations don't.